global/scripts/debug.shlib

```diff
@@ -2,44 +2,271 @@
 #
 #   Function for printing debugging information on caught errors.
 #   It makes use of error trapping. To reset the trap, add
-#   end of the script using "trap - EXIT".
+#   end of the script using "trap - ERR".
 #
 # Input:
 #
 #   — line number
 #   - command ran
 #   - exit status
 
-trap 'on_error $LINENO "$BASH_COMMAND" $?; exit' ERR
+trap 'debug_on_error $LINENO "$BASH_COMMAND" $?; exit' ERR
```

Comment:
This trap contains a hardcoded `exit`, so the debug_disable_exit function doesn't seem like it can do what is described.

Reply:
Removed the exit at trap.